packages/backend-core/src/sql/sql.ts

```diff
@@ -1508,18 +1519,35 @@ class SqlQueryBuilder extends SqlTableQueryBuilder {
     return results.length ? results : [{ [operation.toLowerCase()]: true }]
   }
 
+  private getTableName(
+    table: Table,
+    aliases?: Record<string, string>
+  ): string | undefined {
+    let name = table.name
+    if (
+      table.sourceType === TableSourceType.INTERNAL ||
+      table.sourceId === INTERNAL_TABLE_SOURCE_ID
+    ) {
+      if (!table._id) {
+        return
+      }
+      // SQS uses the table ID rather than the table name
+      name = table._id
+    }
+    return aliases?.[name] ? aliases[name] : name
+  }
```

Comment:
There was logic scattered around a few places on how to determine a table name in the light of SQS being slightly different to other SQL sources, and the potential for table names to be aliased to shorter strings.

I consolidated all of that logic in to this one place and made everywhere* use it.

*not quite, there's a class later in this file that has it's own distinct implementation because it's slightly different